responses/__init__.py

```diff
@@ -1064,14 +1075,16 @@ def _on_request(
                     request, match.get_response(request)
                 )
             except BaseException as response:
-                match.call_count += 1
+                next_index = len(self._calls)
                 self._calls.add(request, response)
+                match.calls.add_call(self._calls[next_index])
```

Comment:
If you have the request and response, why do you need to read out of `self._calls`? Wouldn't `match.calls.add()` work?

Reply:
Actually, that's what I did initially. But then an idea came up:
* reuse an already created call from the "global" list `self._calls`
* have the same call instance in the global list and in the individual response list (`match.calls`)

(using `match.calls.add()` we'll create a separate instance of the same call)

Reply:
Ok, wanting to reuse the objects seems reasonable to me. What do you think of something like

```python
call = Call(request, response)
self._calls.add_call(call)
match.calls.add_call(call)
```

That would save us having to index/count call objects.

Reply:
Looks good, thanks, Let me try...

Reply:
I replaced the index with 

```python
        call = Call(request, response)
        self._calls.add_call(call)
        match.calls.add_call(call)
```

and added an example of usage Response.calls into the README.